app/Http/Controllers/TaskController.php

```diff
@@ -43,62 +45,132 @@ public function index(): InertiaResponse
-                ->orWhere('title', 'LIKE', "%{$query}%")
-                ->orWhere('description', 'LIKE', "%{$query}%");
+            $builder = $this->getTasksMatchingQueryBuilder($builder, $query);
+        }
+
+        if (isset(request()->all()['filter'])) {
+            $builder = $this->getTasksBasedOnFilter($builder, request()->input('filter'));
         }
 
         $tasks = ($builder->count() > 0)
             ? $builder->paginate((int) request()->input('resultsPerPage'))
             : [];
 
         return Inertia::render('Tasks/List', [
-            'fields'            => $fields,
+            'fields'            => Task::$listFields,
+            'filters'           => Task::$filters,
             'links'             => $links,
             'resultsRequestURL' => '/tasks',
             'tasks'             => $tasks
         ]);
     }
 
+    /**
+     * return builder for tasks that match query
+     *
+     * @param Builder $builder
+     * @param string $query
+     * @return Builder
+     */
+    private function getTasksMatchingQueryBuilder(Builder &$builder, string $query): Builder
+    {
+        $userIds = User::where('name', $query)
+            ->orWhere('name', 'LIKE', "%{$query}%")
+            ->pluck('id')
+            ->toArray();
+
+        $statusIds = TaskStatus::where('name', $query)
+            ->orWhere('name', 'LIKE', "%{$query}%")
+            ->pluck('id')
+            ->toArray();
+
+        $priorityIds = TaskPriority::where('name', $query)
+            ->orWhere('name', 'LIKE', "%{$query}%")
+            ->pluck('id')
+            ->toArray();
+
+        $typeIds = TaskType::where('name', $query)
+            ->orWhere('name', 'LIKE', "%{$query}%")
+            ->pluck('id')
+            ->toArray();
+
+        $stateIds = TaskState::where('name', $query)
+            ->orWhere('name', 'LIKE', "%{$query}%")
+            ->pluck('id')
+            ->toArray();
+
+        return $builder->where('id', $query)
+            ->orWhere('title', $query)
+            ->orWhereIn('owner_id', $userIds)
+            ->orWhereIn('author_id', $userIds)
+            ->orWhereIn('status_id', $statusIds)
+            ->orWhereIn('priority_id', $priorityIds)
+            ->orWhereIn('type_id', $typeIds)
+            ->orWhereIn('state_id', $stateIds)
+            ->orWhere('description', $query)
+            ->orWhere('title', 'LIKE', "%{$query}%")
+            ->orWhere('description', 'LIKE', "%{$query}%");
+    }
+
+    /**
+     * return builder for tasks based on filter
+     *
+     * @param Builder $builder
+     * @param string $filter
+     * @return Builder
+     */
+    private function getTasksBasedOnFilter(Builder &$builder, string $filter): Builder
+    {
+        if (strpos($filter, 'Priority') > -1) {
+            $priority = explode(' ', $filter)[0];
+            $priority = TaskPriority::where('name', $priority)->first();
+            return $builder->where('priority_id', $priority->id);
```

Comment:
just on this line add new line before the return. keep it no newline in the if /return blocks.